tests/test_api_project.py

```diff
@@ -133,3 +134,45 @@ def test_project_trigger_create(self):
         t = p.triggers[0]
         self.assertEqual(t.secret_data['secret1'], 'newval')
         self.assertEqual(t.secret_data['secret2'], 'hax0r')
+
+
+    def test_project_trigger_secret_removal(self):
+        project_name = 'projectUpdateTest'
+        trigger_url = f'http://localhost/projects/{project_name}/triggers/'
+        trigger_headers = {'Content-type': 'application/json'}
+        _sign(trigger_url, trigger_headers, 'POST')
+        secret_a = (str(uuid4()), str(uuid4()))  # (key, value) tuple
+        secret_b = (str(uuid4()), str(uuid4()))
+        secret_c = (str(uuid4()), str(uuid4()))
+        secrets = dict([secret_a, secret_b, secret_c])
+        create_data = {
+            'owner': 'tester.testing',
+            'type': TriggerTypes.git_poller.name,
+            **secrets}
+        self.create_projects(project_name)
+        response = self.client.post(
+            trigger_url,
+            headers=trigger_headers,
+            data=json.dumps(create_data))
+        self.assertEqual(201, response.status_code, msg=response.data)
+        actual_triggers = self.get_signed_json(url=trigger_url)
+        self.assertTrue(len(actual_triggers) is 1)
+        actual_trigger = actual_triggers[0]
+        actual_secrets = [
+            secret['name'] for secret in actual_trigger['secrets']]
+        self.assertTrue(len(actual_secrets) is 3)
+        self.assertTrue(secret_a[0] in actual_secrets)
```

Comment:
This is kind of an awkward way to use the assertion helpers. I think things will read more normal with code like:
~~~
 self.assertEqual(3, len(actual_secrets))
 self.assertIn(secret_a[0], actual_secrets)
~~~

Another thing you might consider doing since you have new method devoted to only removing the secret: You could skip all the REST API calls and assertions for creating the trigger and do a smaller, more targeted test with:
~~~
        secret_a = (str(uuid4()), str(uuid4()))  # (key, value) tuple
        secret_b = (str(uuid4()), str(uuid4()))
        secret_c = (str(uuid4()), str(uuid4()))
	self.create_projects('projectName')
	p = Project.query.first()
	pt = ProjectTrigger('user', 'git_poller', p, None, None, secrets)
	db.session.add(pt)
	db.session.commit()

        remove_data = {'secrets': [{'name': secret_b[0], 'value': None}]}
        self.patch_signed_json(
            f"{trigger_url}{pt.id}/", remove_data)

	db.session.refresh(pt)  # I *think* you need this
	self.assertNotIn(secret_b[0], pt.secret_data)
	self.assertIn(secret_a[0], pt.secret_data)
	self.assertIn(secret_c[0], pt.secret_data)
~~~

Reply:
fixed